src/platform_impl/linux/x11/mod.rs

```diff
@@ -407,7 +412,8 @@ impl<T: 'static> EventLoop<T> {
                             super::WindowId::X(wid),
                         )) = event
                         {
-                            wt.redraw_sender.send(wid).unwrap();
+                            wt.redraw_sender.waker.wake().unwrap();
+                            wt.redraw_sender.sender.send(wid).unwrap();
```

Comment:
Shouldn't we send the event first, and only afterwards wake the polling thread? Or is there something here that I'm missing?

This goes for all three places where `Waker::wake()` is used.

Reply:
Yeah, that would make more sense. I'll make the change.